.github/workflows/3rd-party-tests.yml

```diff
@@ -0,0 +1,110 @@
+name: Test on 3rd party libraries
+
+on:
+  push:
+    branches:
+      - "master"
+      - "sqlalchemy_pipeline"
+    paths-ignore:
+      - "docs/*"
+      - "tools/*"
+  workflow_dispatch:
+
+concurrency:
+  # Cancel older requests of the same workflow in the same branch.
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  sqlalchemy:
+    name: sqlalchemy-${{ matrix.sqlalchemy }},psycopg-${{ matrix.impl }},py-${{ matrix.python-version }}
+    # linux should be enough to test if everything works.
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.11"
+          - "3.7"
+        sqlalchemy:
+          - git_main
+          - release
+        impl:
+          - c
+          - python
+
+    env:
+      PSYCOPG_IMPL: ${{ matrix.impl }}
+      DEPS: ./psycopg[test] ./psycopg_pool pytest pytest-xdist
```

Comment:
Out of curiosity, why don't you have a `[test]` extra yourself, specifying `pytest-xdits`? You can then `pip install .[test]` in the `sa` dir to install library and test dependencies.

Reply:
I honestly did not look if sqlalchemy had one.

Maybe we can just use tox here? I'll take a look at what target there are

Reply:
It doesn't seem it has one, no.

TBH Tox has caused me so many problems that doing without it would be preferable for me.

Reply:
looking at the defined env there is one that could maybe be used, that's the github one, but running pytest should be easier